### DISCUSSION TITLE
hash_password function - SQL Server (Potential bug)

### DISCUSSION THREAD

**Discussion by Pieter3033** [sqlpage/SQLPage#116] [Q&A] [answered]
Hi,

I am busy creating a add user page for an admin, not sure if I am doing something wrong.
I am currently struggling converting a input password to `hash_password`, using the built in `sqlpage.hash_password` function.

Password input:
![image](https://github.com/lovasoa/SQLpage/assets/141907966/64f849d9-cfaa-4eb9-b098-72f8c4e6e135)

Password conversion:
![image](https://github.com/lovasoa/SQLpage/assets/141907966/9a964fcb-3b08-46a1-ab3d-35732474ea23)

Database reference:
![image](https://github.com/lovasoa/SQLpage/assets/141907966/691e7baf-6468-4699-bef1-e1615a8ac42c)

Note: The first row (admin) hash_password was generated by myself using argon2. 
The rest of the rows was generated by the hash_password function.
It seems like the input password is not inserted into the function.

Input Tested: 
![image](https://github.com/lovasoa/SQLpage/assets/141907966/88f0de7c-196b-4282-8930-6312a2a3672c)

![image](https://github.com/lovasoa/SQLpage/assets/141907966/9c75b0ff-0795-4bfd-8d85-0a96f794faaf)

**Comment by lovasoa** [comment]
Hello ! Are you using a SQL Server Database ? I recently fixed a bug related to parameter passing to SQL Server. If so, could you test the latest development version ? The easiest way to test it, if you are familiar with docker, is to use the [`main` tag of the official docker image](https://hub.docker.com/r/lovasoa/sqlpage).

If you are not familiar with docker, tell me and I'll generate a download link to the latest development build.

v0.15 should be out in the next few days anyway.

**Reply by Pieter3033** [reply]
Hi,

Yes I am using SQL Server Database. 

Unfortunately I am not familiar with docker.

Regards

**Reply by lovasoa** [reply, marked as answer]
Can you test with v0.15.0-beta.0 ? https://github.com/lovasoa/SQLpage/releases/tag/v0.15.0-beta.0

**Reply by Pieter3033** [reply]
Working perfectly on SQLPage v0.15.0-beta.0, hash_passwords are not truncating anymore.
